### DISCUSSION TITLE
React native: isValid is true when setting defaultsValue despite errors before submitting

### DISCUSSION THREAD

**Discussion by wilav-dev** [react-hook-form/react-hook-form#8488] [Q&A]
### Version Number

7.31.3

### Codesandbox/Expo snack

https://snack.expo.dev/@yamifr07/react-hook-form-v7---controller

### Steps to reproduce

1. Go to [https://snack.expo.dev/@yamifr07/react-hook-form-v7---controller](https://snack.expo.dev/@yamifr07/react-hook-form-v7---controller)
2. Initial state is isValid: true
3. Clear input text
4. Click on Submit
5. Input has an error but form isValid is true instead of false

### Expected behaviour

If an input starts with a valid value but is cleared before submitting, isValid must be false.

### Relevant log output

```shell
{"error":"Field required","isValid":true}
```

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by bluebill1049** [comment]
hey @wilav-dev please refer to the doc for `isValid`, if you are not setting mode to the following then you probably don't need `isValid` state by just checking the errors object. 

<img width="951" alt="Screen Shot 2022-06-10 at 8 52 43 am" src="https://user-images.githubusercontent.com/10513364/172958319-ba6abb42-fb85-45cb-90be-27cb0c153c9f.png">

Note: isValid is affected by mode at [useForm](https://react-hook-form.com/api/useform). This state is only applicable with onChange, onTouched, and onBlur mode.

**Comment by bluebill1049** [comment]
moving into discussion question, feel free to follow up.
